client/src/components/SizeDropDown.jsx

```diff
@@ -8,21 +8,31 @@ class SizeDropDown extends React.Component {
     super(props);
     this.state = {
       dropDownOpen: false,
+      selectedSize: '0',
     };
     this.toggleDropDown = this.toggleDropDown.bind(this);
+    this.handleDropDownSubmit = this.handleDropDownSubmit.bind(this);
   }
 
   toggleDropDown() {
     this.setState({ dropDownOpen: !this.state.dropDownOpen})
   }
 
+  handleDropDownSubmit(event) {
+    // console.log(event.target)
+    this.setState({
+      selectedSize: event.target.value,
+      dropDownOpen: false,
+    })
+  }
+
   render() {
     return (
       <div className={this.props.className}>
-        <DropDownHeader toggleDropDown={this.toggleDropDown}/>
+        <DropDownHeader selectedSize={this.state.selectedSize} toggleDropDown={this.toggleDropDown}/>
         <DropDownList dropDownOpen={this.state.dropDownOpen}>
           {this.props.sizes.map((size) => {
-            return <DropDownItem>{size}</DropDownItem>
+            return <DropDownItem key={size} value={size} onClick={this.handleDropDownSubmit}>{size}</DropDownItem>
```

Comment:
Could use implicit return and convert this to a one-liner if you return to refactor later!

Reply:
ooooo code golf!